Sourcery/Generating/Template/Swift/SwiftTemplate.swift

```diff
@@ -103,39 +99,50 @@ class SwiftTemplate: Template {
                 }
             }
         }
-        return sourceFile.joined(separator: "")
+        let contents = sourceFile.joined(separator: "")
+        let code = "extension TemplateContext {\n\n override func generate() {" + contents + "\n }\n\n}\nrun();"
+        return code
     }
 
-    func render(types: Types, arguments: [String: NSObject]) throws -> String {
-        let context = TemplateContext(types: types, arguments: arguments)
-        let swiftCode = try SwiftTemplate.generateSwiftCode(templateContent: try sourcePath.read(), path: sourcePath)
+    static func loadOrBuild(template: SwiftTemplate, buildDir: Path, cachePath: Path?) throws -> Path {
```

Comment:
There is a separate path for each template, named by template name, so paths are fine now.
I like your idea about using hash as a file name, instead of storing it in archive. It will save us from unneeded unarchiving operations. I guess we should do that for sources cache.

Regarding your last comment we need to store at least a binary, because it takes most of the time to compile generated code. It will be faster when runtime files will be precompiled but it will still take few seconds. With cached binary we just run it. If we combine it with your first suggestion to use hash as a name we can avoid storing `main.swift` and there will be no need in archiving and unarchiving its content at all